src/Microsoft.FeatureManagement/FeatureManager.cs

```diff
@@ -56,12 +69,81 @@ private class ConfigurationCacheItem
 
         public Task<bool> IsEnabledAsync(string feature)
         {
-            return IsEnabledAsync<object>(feature, null, false);
+            return IsEnabledWithVariantsAsync<object>(feature, appContext: null, useAppContext: false, CancellationToken.None);
         }
 
         public Task<bool> IsEnabledAsync<TContext>(string feature, TContext appContext)
         {
-            return IsEnabledAsync(feature, appContext, true);
+            return IsEnabledWithVariantsAsync(feature, appContext, useAppContext: true, CancellationToken.None);
+        }
+
+        public Task<bool> IsEnabledAsync(string feature, CancellationToken cancellationToken)
+        {
+            return IsEnabledWithVariantsAsync<object>(feature, appContext: null, useAppContext: false, cancellationToken);
+        }
+
+        public Task<bool> IsEnabledAsync<TContext>(string feature, TContext appContext, CancellationToken cancellationToken)
+        {
+            return IsEnabledWithVariantsAsync(feature, appContext, useAppContext: true, cancellationToken);
+        }
+
+        private async Task<bool> IsEnabledWithVariantsAsync<TContext>(string feature, TContext appContext, bool useAppContext, CancellationToken cancellationToken)
+        {
+            bool isFeatureEnabled = await IsEnabled(feature, appContext, useAppContext, cancellationToken).ConfigureAwait(false);
+
+            FeatureDefinition featureDefinition = await _featureDefinitionProvider.GetFeatureDefinitionAsync(feature).ConfigureAwait(false);
+
+            if (featureDefinition == null || featureDefinition.Status == FeatureStatus.Disabled)
```

Comment:
I read the original FeatureManager.cs. The previous logic for handling null FeatureDefinition in isEnabledAsync() is:
If IgnoreMissingFeatures option is on, we return false and logWarning.
If IgnoreMissingFeatures option is off, an exception should be thrown.
Should we remain this behavior in IsEnabledWithVariantsAsync()?